tests/phpunit/api/v3/ChainGetSingleTest.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | CiviCRM version 4.6                                                |
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC (c) 2004-2015                                |
+ +--------------------------------------------------------------------+
+ | This file is a part of CiviCRM.                                    |
+ |                                                                    |
+ | CiviCRM is free software; you can copy, modify, and distribute it  |
+ | under the terms of the GNU Affero General Public License           |
+ | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+require_once 'CiviTest/CiviUnitTestCase.php';
+
+/**
+ * This class is intended to test variable leakage in chained getsingle requests
+ *
+ * @package CiviCRM_APIv3
+ */
+class api_v3_ChainGetSingleTest extends CiviUnitTestCase {
+  public $DBResetRequired = FALSE;
+
+  /**
+   * (non-PHPdoc)
+   * @see CiviUnitTestCase::tearDown()
+   */
+  public function tearDown() {
+    $tablesToTruncate = array(
+      'civicrm_campaign',
+      'civicrm_event'
+    );
+    $this->quickCleanup($tablesToTruncate, TRUE);
+  }
+
+  public function testCampaignEvent() {
+    // campaigns are automatically created for each event
+    $event0 = CRM_Core_DAO::createTestObject('CRM_Event_DAO_Event');
+    $event1 = CRM_Core_DAO::createTestObject('CRM_Event_DAO_Event');
+    $event2 = CRM_Core_DAO::createTestObject('CRM_Event_DAO_Event');
+
+    // remove campaign association from second event
+    $event1->campaign_id = 'null';
+    $event1->save();
+
+    $api = $this->callAPIAndDocument('event', 'get', array(
```

Comment:
Happy for @saurabhbatra96 to take a look. Thanks!

--
Frank J. Gómez
Developer & Principal
Ginkgo Street Labs
2121 Decatur Pl NW, Washington, DC 20008
toll free: 888-223-6609

Please send requests for technical support to support@ginkgostreet.com for
the most expeditious response.

On Wed, Oct 7, 2015 at 4:32 PM, Eileen McNaughton <notifications@github.com>
wrote:

> In tests/phpunit/api/v3/ChainGetSingleTest.php
> <https://github.com/civicrm/civicrm-core/pull/6872#discussion_r41442778>:
>
> > +      'civicrm_event'
> > +    );
> > +    $this->quickCleanup($tablesToTruncate, TRUE);
> > +  }
> > +
> > +  public function testCampaignEvent() {
> > +    // campaigns are automatically created for each event
> > +    $event0 = CRM_Core_DAO::createTestObject('CRM_Event_DAO_Event');
> > +    $event1 = CRM_Core_DAO::createTestObject('CRM_Event_DAO_Event');
> > +    $event2 = CRM_Core_DAO::createTestObject('CRM_Event_DAO_Event');
> > +
> > +    // remove campaign association from second event
> > +    $event1->campaign_id = 'null';
> > +    $event1->save();
> > +
> > +    $api = $this->callAPIAndDocument('event', 'get', array(
>
> @GinkgoFJG <https://github.com/GinkgoFJG> @saurabhbatra96
> <https://github.com/saurabhbatra96> was going to take a look - if you are
> going to fix please confirm so he doesn't duplicate
>
> I expect this function
> https://github.com/civicrm/civicrm-core/blob/81621fee4d9c2e8c6f5272a1c6b38a86186fe185/Civi/API/Subscriber/ChainSubscriber.php#L89-89
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/civicrm/civicrm-core/pull/6872/files#r41442778>.
>